File: src/main/java/org/apache/fineract/client/services/SelfSavingsAccountApi.java
/**
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements. See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership. The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License. You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied. See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
package org.apache.fineract.client.services;

import org.apache.fineract.client.CollectionFormats.*;

import rx.Observable;
import retrofit2.http.*;

import okhttp3.RequestBody;
import okhttp3.ResponseBody;
import okhttp3.MultipartBody;

import org.apache.fineract.client.models.GetSelfSavingsAccountsAccountIdChargesResponse;
import org.apache.fineract.client.models.GetSelfSavingsAccountsAccountIdChargesSavingsAccountChargeIdResponse;
import org.apache.fineract.client.models.GetSelfSavingsAccountsAccountIdTransactionsTransactionIdResponse;
import org.apache.fineract.client.models.GetSelfSavingsAccountsResponse;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public interface SelfSavingsAccountApi {
  /**
   * 
   * 
   * @param accountId  (required)
   * @param command  (optional)
   * @param body  (optional)
   * @return Observable&lt;String&gt;
   */
  @Headers({
    "Content-Type:application/json"
  })
  @PUT("self/savingsaccounts/{accountId}")
  Observable<String> modifySavingsAccountApplication(
    @retrofit2.http.Path("accountId") Long accountId, @retrofit2.http.Query("command") String command, @retrofit2.http.Body String body
  );

  /**
   * List Savings Charges
   * Lists Savings Charges  Example Requests:  self/savingsaccounts/1/charges  self/savingsaccounts/1/charges?chargeStatus&#x3D;inactive  self/savingsaccounts/1/charges?fields&#x3D;name,amountOrPercentage
   * @param accountId accountId (required)
   * @param chargeStatus chargeStatus (optional, default to &quot;all&quot;)
   * @return Observable&lt;List&lt;GetSelfSavingsAccountsAccountIdChargesResponse&gt;&gt;
   */
  @GET("self/savingsaccounts/{accountId}/charges")
  Observable<List<GetSelfSavingsAccountsAccountIdChargesResponse>> retrieveAllSavingsAccountCharges1(
    @retrofit2.http.Path("accountId") Long accountId, @retrofit2.http.Query("chargeStatus") String chargeStatus
  );

  /**
   * Retrieve a savings account
   * Retrieves a savings account  Example Requests :  self/savingsaccounts/1   self/savingsaccounts/1?associations&#x3D;transactions
   * @param accountId accountId (required)
   * @param chargeStatus chargeStatus (optional, default to &quot;all&quot;)
   * @return Observable&lt;GetSelfSavingsAccountsResponse&gt;
   */
  @GET("self/savingsaccounts/{accountId}")
  Observable<GetSelfSavingsAccountsResponse> retrieveSavings(
    @retrofit2.http.Path("accountId") Long accountId, @retrofit2.http.Query("chargeStatus") String chargeStatus
  );

  /**
   * Retrieve a Savings account Charge
   * Retrieves a Savings account Charge  Example Requests:  self/savingsaccounts/1/charges/5   self/savingsaccounts/1/charges/5?fields&#x3D;name,amountOrPercentage
   * @param accountId accountId (required)
   * @param savingsAccountChargeId savingsAccountChargeId (required)
   * @return Observable&lt;GetSelfSavingsAccountsAccountIdChargesSavingsAccountChargeIdResponse&gt;
   */
  @GET("self/savingsaccounts/{accountId}/charges/{savingsAccountChargeId}")
  Observable<GetSelfSavingsAccountsAccountIdChargesSavingsAccountChargeIdResponse> retrieveSavingsAccountCharge1(
    @retrofit2.http.Path("accountId") Long accountId, @retrofit2.http.Path("savingsAccountChargeId") Long savingsAccountChargeId
  );

  /**
   * Retrieve Savings Account Transaction
   * Retrieves Savings Account Transaction  Example Requests:  self/savingsaccounts/1/transactions/1
   * @param accountId accountId (required)
   * @param transactionId transactionId (required)
   * @return Observable&lt;GetSelfSavingsAccountsAccountIdTransactionsTransactionIdResponse&gt;
   */
  @GET("self/savingsaccounts/{accountId}/transactions/{transactionId}")
  Observable<GetSelfSavingsAccountsAccountIdTransactionsTransactionIdResponse> retrieveSavingsTransaction(
    @retrofit2.http.Path("accountId") Long accountId, @retrofit2.http.Path("transactionId") Long transactionId
  );

  /**
   * 
   * 
   * @param command  (optional)
   * @param body  (optional)
   * @return Observable&lt;String&gt;
   */
  @Headers({
    "Content-Type:application/json"
  })
  @POST("self/savingsaccounts")
  Observable<String> submitSavingsAccountApplication(
    @retrofit2.http.Query("command") String command, @retrofit2.http.Body String body
  );

  /**
   * 
   * 
   * @param clientId  (optional)
   * @param productId  (optional)
   * @return Observable&lt;String&gt;
   */
  @GET("self/savingsaccounts/template")
  Observable<String> template18(
    @retrofit2.http.Query("clientId") Long clientId, @retrofit2.http.Query("productId") Long productId
  );

}

File: src/main/kotlin/org/openapitools/client/apis/SelfSavingsAccountApi.kt
package org.openapitools.client.apis

import org.openapitools.client.models.GetSelfSavingsAccountsAccountIdChargesResponse
import org.openapitools.client.models.GetSelfSavingsAccountsAccountIdChargesSavingsAccountChargeIdResponse
import org.openapitools.client.models.GetSelfSavingsAccountsAccountIdTransactionsTransactionIdResponse
import org.openapitools.client.models.GetSelfSavingsAccountsResponse
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.http.PUT
import retrofit2.http.Path
import retrofit2.http.Query

interface SelfSavingsAccountApi {
    /**
     *
     *
     * Responses:
     *  - 0: default response
     *
     * @param accountId
     * @param command  (optional)
     * @param body  (optional)
     * @return [kotlin.String]
     */
    @PUT("v1/self/savingsaccounts/{accountId}")
    suspend fun modifySavingsAccountApplication(
        @Path("accountId") accountId: Long,
        @Query("command") command: String? = null,
        @Body body: String? = null
    ): String

    /**
     * List Savings Charges
     * Lists Savings Charges  Example Requests:  self/savingsaccounts/1/charges  self/savingsaccounts/1/charges?chargeStatus&#x3D;inactive  self/savingsaccounts/1/charges?fields&#x3D;name,amountOrPercentage
     * Responses:
     *  - 200: OK
     *
     * @param accountId accountId
     * @param chargeStatus chargeStatus (optional, default to "all")
     * @return [kotlin.collections.List<GetSelfSavingsAccountsAccountIdChargesResponse]
     */
    @GET("v1/self/savingsaccounts/{accountId}/charges")
    suspend fun retrieveAllSavingsAccountCharges1(
        @Path("accountId") accountId: Long,
        @Query("chargeStatus") chargeStatus: String? = "all"
    ): List<GetSelfSavingsAccountsAccountIdChargesResponse>

    /**
     * Retrieve a savings account
     * Retrieves a savings account  Example Requests :  self/savingsaccounts/1   self/savingsaccounts/1?associations&#x3D;transactions
     * Responses:
     *  - 200: OK
     *
     * @param accountId accountId
     * @param chargeStatus chargeStatus (optional, default to "all")
     * @return [GetSelfSavingsAccountsResponse]
     */
    @GET("v1/self/savingsaccounts/{accountId}")
    suspend fun retrieveSavings(
        @Path("accountId") accountId: Long,
        @Query("chargeStatus") chargeStatus: String? = "all"
    ): GetSelfSavingsAccountsResponse

    /**
     * Retrieve a Savings account Charge
     * Retrieves a Savings account Charge  Example Requests:  self/savingsaccounts/1/charges/5   self/savingsaccounts/1/charges/5?fields&#x3D;name,amountOrPercentage
     * Responses:
     *  - 200: OK
     *
     * @param accountId accountId
     * @param savingsAccountChargeId savingsAccountChargeId
     * @return [GetSelfSavingsAccountsAccountIdChargesSavingsAccountChargeIdResponse]
     */
    @GET("v1/self/savingsaccounts/{accountId}/charges/{savingsAccountChargeId}")
    suspend fun retrieveSavingsAccountCharge1(
        @Path("accountId") accountId: Long,
        @Path("savingsAccountChargeId") savingsAccountChargeId: Long
    ): GetSelfSavingsAccountsAccountIdChargesSavingsAccountChargeIdResponse

    /**
     * Retrieve Savings Account Transaction
     * Retrieves Savings Account Transaction  Example Requests:  self/savingsaccounts/1/transactions/1
     * Responses:
     *  - 200: OK
     *
     * @param accountId accountId
     * @param transactionId transactionId
     * @return [GetSelfSavingsAccountsAccountIdTransactionsTransactionIdResponse]
     */
    @GET("v1/self/savingsaccounts/{accountId}/transactions/{transactionId}")
    suspend fun retrieveSavingsTransaction(
        @Path("accountId") accountId: Long,
        @Path("transactionId") transactionId: Long
    ): GetSelfSavingsAccountsAccountIdTransactionsTransactionIdResponse

    /**
     *
     *
     * Responses:
     *  - 0: default response
     *
     * @param command  (optional)
     * @param body  (optional)
     * @return [kotlin.String]
     */
    @POST("v1/self/savingsaccounts")
    suspend fun submitSavingsAccountApplication(
        @Query("command") command: String? = null,
        @Body body: String? = null
    ): String

    /**
     *
     *
     * Responses:
     *  - 0: default response
     *
     * @param clientId  (optional)
     * @param productId  (optional)
     * @return [kotlin.String]
     */
    @GET("v1/self/savingsaccounts/template")
    suspend fun template18(
        @Query("clientId") clientId: Long? = null,
        @Query("productId") productId: Long? = null
    ): String

}
